### DISCUSSION TITLE
Create a unified error boundary strategy

### DISCUSSION THREAD

**Discussion by KevinMGranger** [dora-metrics/pelorus#808] [General]
We sometimes see exceptions that bubble up to the top of the exporter and crash it. Sometimes it's from code processing a metric, sometimes it's from the client we're using to talk to a certain backend, etc.

We should discuss and find a way to contain these errors. An error with a single metric should be logged, and the metric should be skipped. At the same time, we can't let _all_ unknown exceptions be swallowed, because then we hide ones we need to now about. (No [Pokémon exception handling](https://wiki.c2.com/?PokemonExceptionHandling) here!)

**Comment by mpryc** [comment]
Moving to discussion.
